src/FiberMap.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace React\Async;
+
+use React\Promise\PromiseInterface;
+
+/**
+ * @internal
+ */
+final class FiberMap
+{
+    private static array $status = [];
+    private static array $map   = [];
+
+    public static function register(\Fiber $fiber): void
+    {
+        self::$status[\spl_object_hash($fiber)] = false;
+        self::$map[\spl_object_hash($fiber)] = [];
```

Comment:
Have you considered using `spl_object_id()` instead of `spl_object_hash()`? (https://github.com/reactphp/async/pull/20#discussion_r817120003)

It's my understanding this would return much "shorter" identifiers and thus save some memory when you have thousands of fibers/promises in this map. Not sure how relevant this is, might as well be a micro-optimization at this point, but just wanted to bring this up.

Reply:
@clue Hmm whoops I thought I made that change 0_o.